### PR TITLE
SLING-9061  - Adding support for evaluating the origin as well as the referrer and pluggable referrer configurations

### PR DESCRIPTION
**Pull request by klcodanr**

### CODE REVIEW

File: src/main/java/org/apache/sling/security/impl/ReferrerFilter.java [hunk 1]
```diff
@@ -260,16 +264,32 @@ private Pattern[] createRegexPatterns(final String[] regexps) {
         return patterns.toArray(new Pattern[0]);
     }
 
+    private String[] mergeValues(String[] primary, List<ReferrerFilterAmendment> amendments,
```
Comment (kwin) [inline src/main/java/org/apache/sling/security/impl/ReferrerFilter.java:267]: why not returning a collection here? IIUC it is later on anyways converted to a list

Comment (klcodanr) [reply src/main/java/org/apache/sling/security/impl/ReferrerFilter.java]: Honestly, minimizing changes, but collections are fine or needed 4/5 times so it makes sense to change.

**Comment by klcodanr** [comment]
@kwin / @cziegeler / @anchela 

Any further feedback? If not I'll merge this in 24 hours.

**Comment by anchela** [comment]
@klcodanr , there was public holiday in switzerland..... sorry for not responding.

**Comment by klcodanr** [comment]
Ah sorry for the rush @anchela. Hope you had a great holiday!

**Pull request merged**